### PR TITLE
#600 PsBasic.Default constructor must only set final variables and call other constructors

### PR DESCRIPTION
**Pull request by xupyprmv**
#600 I moved code from constructor to separate method. I added `@SuppressWarnings("PMD.TooManyMethods")` annotation to class.
Also I added test that uses PsBasic.Default to be sure that PsBasic.Default works fine.

### CODE REVIEW

**Comment by davvd** [comment]
@xupyprmv Thanks, let me find someone who can review this pull request

**Comment by davvd** [comment]
@hdouss this pull request is for you, please review

File: src/test/java/org/takes/facets/auth/PsBasicTest.java [hunk 1]
```diff
@@ -81,6 +87,42 @@ public void handleConnectionWithValidCredential() throws Exception {
...
+                String.format("%s %s urn:basic:%s", user, password, user)
+            )
+        ).enter(
+            new RqWithHeaders(
+                new RqFake(
+                    RqMethod.GET,
+                    String.format(
+                        PsBasicTest.VALID_CODE,
+                        // @checkstyle MagicNumberCheck (1 line)
+                        RandomStringUtils.randomAlphanumeric(10)
```
Comment (HDouss) [inline src/test/java/org/takes/facets/auth/PsBasicTest.java]: @xupyprmv Let's use Tv.TEN instead.

**Comment by HDouss** [comment]
@xupyprmv One minor remark.

**Comment by xupyprmv** [comment]
@HDouss Thanks for CR. I updated PR according to your remark. Please continue.

**Comment by HDouss** [comment]
@xupyprmv Thanks.

**Comment by HDouss** [comment]
@rultor merge.

**Comment by xupyprmv** [comment]
@yegor256 Please confirm this PR.

**Comment by yegor256** [comment]
@rultor try to merge

**Comment by davvd** [comment]
@ypshenychka please, review this task for compliance with our [quality rules](http://at.teamed.io/qa.html)

**Comment by davvd** [comment]
@rultor deploy now pls

**Comment by ypshenychka** [comment]
@HDouss 
According to our [QA Rules](http://at.teamed.io/qa.html):
>The code reviewer found at least three problems in the code.

Only one issue was found during code review. 
Please confirm that you'll try to find at least three problems while future reviews.

**Comment by HDouss** [comment]
>Please confirm that you'll try to find at least three problems while future reviews.

@ypshenychka I confirm.

**Comment by ypshenychka** [comment]
@HDouss Thanks.

**Comment by ypshenychka** [comment]
@davvd Quality is acceptable.

**Comment by davvd** [comment]
> @davvd Quality is acceptable.

@ypshenychka thanks for the QA review, we'll work better next time

**Comment by davvd** [comment]
@hdouss added 10 mins to @ypshenychka, for QA review, payment code is `80010305`. thank you, added **15 mins** to your acc, payment num is `80010346`, 158 hours and 55 mins spent in total to complete. +15 added to your rating, at the moment it is: [+3758](http://www.netbout.com/b/36032?open=rating)

**Pull request merged**